From 4e7ff267147ad3a62953c8e81a49e6289cff58be Mon Sep 17 00:00:00 2001
From: Edward Wang <ewang@cyngn.com>
Date: Thu, 9 Apr 2015 16:06:02 -0700
Subject: [PATCH 27/74] FM: retune last frequency when resume after call

FM Activity may not available to send down frequency again
after phone call. Then FM service need retune saved frequency.

Change-Id: I26f0921bd8878331c3dedc6e8ff96873dd5a84a6
(cherry picked from commit 4b86c869026d6ef7504d2cecca4c3db9fa9d8c4b)
---
 fmapp2/src/com/caf/fmradio/FMRadioService.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fmapp2/src/com/caf/fmradio/FMRadioService.java b/fmapp2/src/com/caf/fmradio/FMRadioService.java
index 78bf9ab..c1cedab 100644
--- a/fmapp2/src/com/caf/fmradio/FMRadioService.java
+++ b/fmapp2/src/com/caf/fmradio/FMRadioService.java
@@ -144,6 +144,7 @@ public class FMRadioService extends Service
    private int mServiceStartId = -1;
    private boolean mServiceInUse = false;
    private static boolean mMuted = false;
+   private static int mFreq = 0;
    private static boolean mResumeAfterCall = false;
    private static String mAudioDevice="headset";
    MediaRecorder mRecorder = null;
@@ -1590,6 +1591,8 @@ public class FMRadioService extends Service
                 } catch (RemoteException e) {
                     e.printStackTrace();
                 }
+            } else if (mFreq > 0) {
+                tune(mFreq);
             }
         }
     }
@@ -2938,6 +2941,7 @@ public class FMRadioService extends Service
       {
          mReceiver.setStation(frequency);
          bCommandSent = true;
+         mFreq = frequency;
       }
       return bCommandSent;
    }
-- 
2.17.1

